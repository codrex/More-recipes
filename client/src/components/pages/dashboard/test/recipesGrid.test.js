import React from 'react';
import RecipesGrid from '../recipesGrid';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

const recipe = {
  recipeName: 'coco rice',
  category: 'breakfast',
  views: 22,
  upVotes: 22,
  downVotes: 0,
  id: 1
};

describe('Recipes grid component ', () => {
  test('render as expected without any recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipesGrid recipes={[]} />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render as expected with one recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipesGrid recipes={[recipe]} />
      </BrowserRouter>
    );
    // const componentTwo = renderer.create(<RecipesGrid recipes={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
