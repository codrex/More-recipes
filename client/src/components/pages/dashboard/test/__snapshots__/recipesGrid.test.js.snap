// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes grid component  render as expected with one recipe 1`] = `
<div
  className="col-12 grid"
>
  <div
    className="recipe-card"
  >
    <div
      className="recipe-card-body"
    >
      <img
        alt="coco rice"
        className="recipe-card-img"
        src="https://i.pinimg.com/564x/b5/be/08/b5be084a849fa67c190cc85b1e9216c3.jpg"
      />
      <div
        className="recipe-card-underlay"
      >
        <p
          className="lead"
        >
          breakfast
        </p>
        <h4
          className="recipe-card-title"
        >
          coco rice
        </h4>
        <a
          href="recipe/1"
          onClick={[Function]}
        >
          <button
            className="btn text-capitalize recipe-card-underlay-btn"
            data-dismiss="modal"
            data-target={undefined}
            data-toggle={undefined}
            onClick={undefined}
            type="button"
          >
            view details
          </button>
        </a>
      </div>
    </div>
    <div
      className="recipe-card-footer"
    >
      <div
        className="icon-wrapper"
        onClick={undefined}
        onMouseEnter={undefined}
      >
        <i
          className="icon fa fa-thumbs-up recipe-card-icon "
          data-target={undefined}
          data-toggle={undefined}
        >
          <span>
            22
          </span>
        </i>
      </div>
      <div
        className="icon-wrapper"
        onClick={undefined}
        onMouseEnter={undefined}
      >
        <i
          className="icon fa fa-thumbs-down recipe-card-icon "
          data-target={undefined}
          data-toggle={undefined}
        >
          <span>
            0
          </span>
        </i>
      </div>
      <div
        className="icon-wrapper"
        onClick={undefined}
        onMouseEnter={undefined}
      >
        <i
          className="icon fa fa-eye recipe-card-icon"
          data-target={undefined}
          data-toggle={undefined}
        >
          <span>
            22
          </span>
        </i>
      </div>
      <div
        className="icon-wrapper pointer"
        onClick={undefined}
        onMouseEnter={undefined}
      >
        <i
          className="icon fa fa-heart-o recipe-card-icon fav-color"
          data-target={undefined}
          data-toggle={undefined}
        >
          <span />
        </i>
      </div>
    </div>
  </div>
</div>
`;

exports[`Recipes grid component  render as expected without any recipe 1`] = `
<div
  className="col-12 grid"
>
  <div
    className="display-3 text-uppercase msg-display no-margin"
  >
    <h1>
      Sorry..
      <br />
      No recipe was found.
    </h1>
  </div>
</div>
`;
